app/src/main/java/com/mashup/mobalmobal/ui/sign/signin/SignInFragment.kt

```diff
@@ -1,25 +1,147 @@
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        setGoogleLogin()
+        setFacebookLogin()
+
+        binding.signInGoogle.setOnClickListener {
+            onGoogleClicked()
+        }
+
+        viewModel.showToast.observe(viewLifecycleOwner, { message ->
+            showToastMessage(message)
+        })
+    }
+
+    override fun onDestroyView() {
+        super.onDestroyView()
+        _binding = null
+    }
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+        callbackManager.onActivityResult(requestCode, resultCode, data)
+    }
+
+    private fun setGoogleLogin() {
+        val gso = GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
+            .requestIdToken(getString(R.string.default_web_client_id))
+            .requestEmail()
+            .build()
+
+        googleSignInClient = GoogleSignIn.getClient(requireActivity(), gso)
+
+        googleLoginLauncher = registerForActivityResult(GoogleLoginResultContract()) { idToken ->
+            idToken?.let {
+                viewModel.handleGoogleAccessToken(it)
+            }
+        }
+    }
+
+    private fun setFacebookLogin() {
+        callbackManager = CallbackManager.Factory.create()
+
+        binding.signInFacebook.setPermissions("email", "public_profile")
+        binding.signInFacebook.fragment = this
+        binding.signInFacebook.registerCallback(
+            callbackManager,
+            object : FacebookCallback<LoginResult> {
+                override fun onSuccess(result: LoginResult?) {
+                    result?.let {
+                        viewModel.handleFacebookAccessToken(it.accessToken)
+                    }
+                }
+
+                override fun onCancel() {
+                }
+
+                override fun onError(error: FacebookException?) {
+                    error?.let {
+                        showToastMessage(it.message!!)
+                    }
+                }
+            })
+    }
+
+    private fun onGoogleClicked() {
+        googleLoginLauncher.launch(googleSignInClient.signInIntent)
     }
 
     fun navigateSignInToSignUp() =
         findNavController().navigate(R.id.action_signInFragment_to_signUpFragment)
-}
\ No newline at end of file
+
+    inner class GoogleLoginResultContract : ActivityResultContract<Intent, String>() {
+        override fun createIntent(context: Context, input: Intent): Intent = input
+
+        override fun parseResult(resultCode: Int, intent: Intent?): String {
+            val task = GoogleSignIn.getSignedInAccountFromIntent(intent)
+
+            try {
+                val account = task.getResult(ApiException::class.java)
+
+                account?.let {
+                    return account.idToken!!
+                }
```

Comment:
여기도 !! 없애주세요! 차라리 throw 해주세요
```suggestion
                return account?.idToken ?: throw IllegalStateException("Account token must be not null")
```
그리고 밑에 catch에서 예외처리 해주세요!